### PR TITLE
Router: improve handling of `binpack=oneline`

### PR DESCRIPTION
**Pull request by kitbellew**

### CODE REVIEW

File: scalafmt-tests/src/test/resources/scalajs/Apply.stat [hunk 1]
```diff
@@ -181,3 +181,295 @@ object a {
...
+===
+object a {
+  new SimpleMethodName(new SimpleMethodName(new SimpleMethodName(
+    validateSimpleEncodedName(name, 0, len, openAngleBracketOK = false)))).foo
+}
+>>>
+object a {
+  new SimpleMethodName(
+      new SimpleMethodName(new SimpleMethodName(validateSimpleEncodedName(name,
+                  0, len, openAngleBracketOK = false)))
```
Comment (tgodzik) [inline scalafmt-tests/src/test/resources/scalajs/Apply.stat:279]: This indentation seems excessive, shouldn't it just be 4? Looks like a lot of other test cases have double indentation. This even seem 3x (12 spaces)

Comment (kitbellew) [reply scalafmt-tests/src/test/resources/scalajs/Apply.stat:279]: it only looks strange with binpacking. each open paren adds an indentation. there is an indent once setting, for those who like this type of [un]formatting.

Comment (tgodzik) [reply scalafmt-tests/src/test/resources/scalajs/Apply.stat:279]: Ach, it makes sense! Thanks for explaining.

**Review by tgodzik — APPROVED** [review]
LGTM!

**Pull request merged**